pkg/build/controller/image_change_controller.go

```diff
@@ -54,6 +48,8 @@ func (c *ImageChangeController) HandleImageRepo() {
 				continue
 			}
 			icTrigger := trigger.ImageChange
+			// TODO this will currently match imagerepos with the same name but in a different
+			// namespace, we should be checking that the namespaces match.
 			if icTrigger.From.Name != imageRepo.Name {
```

Comment:
didn't stop to process your objectref changes and see that it could be fixed at this point, wasn't in scope for what i was trying to do, just wanted to make a note that it needed to be fixed so we didn't lose track of it.

will fix it.